obspy/core/stream.py

```diff
@@ -795,13 +795,13 @@ def getGaps(self, min_gap=None, max_gap=None):
                 continue
             # different sampling rates should always result in a gap or overlap
             if self.traces[_i].stats.delta == self.traces[_i + 1].stats.delta:
-                flag = True
+                same_sampling_rate = True
             else:
-                flag = False
+                same_sampling_rate = False
             stats = self.traces[_i].stats
             stime = stats['endtime']
             etime = self.traces[_i + 1].stats['starttime']
-            delta = etime.timestamp - stime.timestamp
+            delta = etime.timestamp - (stime.timestamp + stats.delta)
```

Comment:
> My line of thought is that the earlier trace "really" ends not at the time of last sample, but delta after it -- at least of you think of a trace starting exactly at first sample time. Kind of a subjective thing I guess..

Yea..but you kind of shift the start time of the next trace a sample earlier ;-) But as you said: its subjective and the end result is identical :-)